Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -125,7 +127,13 @@ async function main() {
     finally {
         if(kuduServiceUtility) {
             await addReleaseAnnotation(azureEndpoint, appService, isDeploymentSuccess);
-            await kuduServiceUtility.updateDeploymentStatus(isDeploymentSuccess, null, {'type': 'Deployment', slotName: appService.getSlot()});
+            let activeDeploymentID: string = await kuduServiceUtility.updateDeploymentStatus(isDeploymentSuccess, null, {'type': 'Deployment', slotName: appService.getSlot()});
+            if(zipDeploymentID && activeDeploymentID && isDeploymentSuccess) {
```

Comment:
Why is there a check for isDeploymentSuccess? Shouldn't we be updating the status irrespective of success or failure?

Reply:
The above line `kuduServiceUtility.updateDeploymentStatus` updates the deployment status despite status. Here, we copy the manifest file from ZIP deploy ID directory to VSTS Deploy ID directory, only in success case, as failure deployment wont have any manifest file.